src/main/java/org/folio/circulation/resources/DueDateNotRealTimeScheduledNoticeProcessingResource.java

```diff
@@ -0,0 +1,84 @@
+import org.folio.circulation.domain.notice.schedule.ScheduledNotice;
+import org.folio.circulation.domain.notice.schedule.ScheduledNoticeGroupDefinition;
+import org.folio.circulation.domain.notice.schedule.ScheduledNoticesRepository;
+import org.folio.circulation.domain.notice.schedule.TriggeringEvent;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.CqlSortClause;
+import org.folio.circulation.support.CqlSortBy;
+import org.folio.circulation.support.Result;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.LocalDate;
+import org.joda.time.LocalTime;
+
+import io.vertx.core.http.HttpClient;
+
+public class DueDateNotRealTimeScheduledNoticeProcessingResource extends ScheduledNoticeProcessingResource {
+
+  public DueDateNotRealTimeScheduledNoticeProcessingResource(HttpClient client) {
+    super("/circulation/due-date-not-real-time-scheduled-notices-processing", client);
+  }
+
+  @Override
+  protected CompletableFuture<Result<MultipleRecords<ScheduledNotice>>> findNoticesToSend(
+    ScheduledNoticesRepository scheduledNoticesRepository, int limit) {
+
+    List<CqlSortClause> cqlSortClauses = Stream.of(
+      "recipientUserId", "noticeConfig.templateId",
+      "triggeringEvent", "noticeConfig.format",
+      "noticeConfig.timing")
+      .map(CqlSortClause::ascending).collect(Collectors.toList());
+
+    DateTime timeLimit = LocalDate.now().toDateTime(LocalTime.MIDNIGHT);
+    return scheduledNoticesRepository.findNotices(timeLimit, false,
+      Collections.singletonList(TriggeringEvent.DUE_DATE),
+      CqlSortBy.sortBy(cqlSortClauses), limit);
+  }
+
+  @Override
+  protected CompletableFuture<Result<MultipleRecords<ScheduledNotice>>> handleNotices(
+    Clients clients, MultipleRecords<ScheduledNotice> notices) {
+
+    final DueDateNotRealTimeScheduledNoticeHandler dueDateNoticeHandler =
+      DueDateNotRealTimeScheduledNoticeHandler.using(clients, DateTime.now(DateTimeZone.UTC));
+
+    Map<ScheduledNoticeGroupDefinition, List<ScheduledNotice>> orderedGroups =
+      notices.getRecords().stream().collect(Collectors.groupingBy(
+        ScheduledNoticeGroupDefinition::from,
+        LinkedHashMap::new,
+        Collectors.toList()));
+
+    boolean fetchedAllTheRecords = notices.getTotalRecords().equals(notices.getRecords().size());
+    //If not all the records are fetched then the last group is cut off because there might be only a part of it
+    //If there is only one group, it is taken into processing
+    int limit = fetchedAllTheRecords
+      ? orderedGroups.size()
+      : max(orderedGroups.size() - 1, 1);
+
+    List<Pair<ScheduledNoticeGroupDefinition, List<ScheduledNotice>>> noticeGroups =
+      orderedGroups.entrySet()
+        .stream()
+        .limit(limit)
+        .map(e -> Pair.of(e.getKey(), e.getValue()))
+        .collect(Collectors.toList());
+
+    return dueDateNoticeHandler.handleNotices(noticeGroups)
```

Comment:
@dmytrotur, why we cannot use map here, and using list of pairs instead?

Reply:
@bohdan-suprun Thanks for pointing that. I just realized we don't need a list of pairs here, it enough to have a list of lists.

Reply:
Done

Reply:
Thanks @dmytrotur